lib/modules/manager/hermit/artifacts.ts

```diff
@@ -0,0 +1,256 @@
+import pMap from 'p-map';
+import upath from 'upath';
+import { logger } from '../../../logger';
+import { exec } from '../../../util/exec';
+import type { ExecOptions } from '../../../util/exec/types';
+import {
+  localPathIsSymbolicLink,
+  readLocalFile,
+  readLocalSymlink,
+} from '../../../util/fs';
+import { getRepoStatus } from '../../../util/git';
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+import type {
+  HermitPackageDependency,
+  ReadContentResult,
+  UpdateHermitError,
+} from './types';
+
+/**
+ * updateArtifacts runs hermit install for each updated dependencies
+ */
+export async function updateArtifacts({
+  updatedDeps,
+  packageFileName,
+}: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
+  logger.debug({ packageFileName }, `hermit.updateArtifacts()`);
+
+  try {
+    for (const dep of updatedDeps) {
+      await updateHermitPackage({ ...dep, packageFileName });
+    }
+  } catch (err) {
+    const execErr: UpdateHermitError<any> = err;
+    logger.warn(
+      { stdout: execErr.stdout, stderr: execErr.stderr },
+      `error updating hermit packages.`
+    );
+    return [execErr.from, execErr.to].map((lockFile) => ({
+      artifactError: {
+        lockFile,
+        stderr: execErr.stderr,
+      },
+    }));
+  }
+
+  logger.debug(`scanning the changes after update`);
+
+  const updateResult = await getUpdateResult(packageFileName);
+
+  logger.debug({ updateResult }, `update result for hermit`);
+
+  return updateResult;
+}
+
+/**
+ * getContent returns the content of either link or a normal file
+ */
+async function getContent(file: string): Promise<ReadContentResult> {
+  let contents: string | null = '';
+  const isSymlink = await localPathIsSymbolicLink(file);
+  if (isSymlink) {
+    contents = await readLocalSymlink(file);
+  } else {
+    contents = await readLocalFile(file, 'utf8');
```

Comment:
are those files all text files? i though they're compressed tar files?

Reply:
actually there will only be symlinks, not other type of files. will make the change instead.